### PR TITLE
CDPS-373: Fixing the location history links

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: server/controllers/converters/locationDetailsConverter.test.ts [hunk 1]
```diff
@@ -54,70 +83,20 @@ describe('locationDetailsConverter', () => {
...
         locationDetails: [
-          { ...expectedLocationDetailsForDisplay, livingUnitId: 1 },
-          { ...expectedLocationDetailsForDisplay, livingUnitId: 2 },
+          { ...expectedLocationDetailsForDisplay, location: '1-1-1' },
+          { ...expectedLocationDetailsForDisplay, location: '1-1-2' },
         ],
       })
     })
-
-    it('converts location details without staff details', () => {
```
Comment (brightonsbox) [inline server/controllers/converters/locationDetailsConverter.test.ts:72]: Removing accidentally copy / pasted tests

File: server/interfaces/pages/locationDetailsPageData.ts [hunk 2]
```diff
@@ -4,14 +4,12 @@ export interface LocationDetailsOccupant {
 }
 
 export interface LocationDetailsForDisplay {
-  agencyId: string
```
Comment (brightonsbox) [inline server/interfaces/pages/locationDetailsPageData.ts:7]: Can now remove a few of these from the interface

File: server/controllers/converters/locationDetailsConverter.ts [hunk 3]
```diff
@@ -3,38 +3,57 @@ import {
...
+    agencyName: groupedLocations.agencyName,
+    fromDate: formatDate(groupedLocations.fromDate, 'short'),
+    toDate: formatDate(groupedLocations.toDate, 'short') || 'Unknown',
+    locationDetails: groupedLocations.locationDetails.map(location =>
+      convertLocationDetails(location, defaultLocationHistoryToDate),
+    ),
   }
 }
+
+function generateLocationHistoryLink(location: LocationDetails, defaultLocationHistoryToDate: Date): string {
```
Comment (brightonsbox) [inline server/controllers/converters/locationDetailsConverter.ts:47]: Moved this out of the nunjucks template and into here

File: server/controllers/converters/locationDetailsConverter.ts [hunk 4]
```diff
@@ -3,38 +3,64 @@ import {
...
 }
+
+function generateLocationHistoryLink(
+  location: LocationDetails,
+  prisonerNumber: string,
+  defaultLocationHistoryToDate: Date,
+): string {
+  if (location.isTemporaryLocation || !location.assignmentDateTime) return null
+
+  const locationHistoryLink = new URL(`/prisoner/${prisonerNumber}/location-history`, config.serviceUrls.digitalPrison)
```
Comment (davidmiddletonmoj) [inline server/controllers/converters/locationDetailsConverter.ts:57]: Should this not be the location history page in the profile service?
Or is that page not ready to "go live" yet?

**Review by davidmiddletonmoj — APPROVED** [review]

**Pull request merged**